src/sync/node.js

```diff
@@ -0,0 +1,65 @@
+import PushManagerFactory from './PushManager';
+import { PUSH_DISCONNECT, PUSH_CONNECT } from './constants';
+import FullProducerFactory from '../producer';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-sync:sync-manager');
+
+/**
+ * Factory of SyncManager for node
+ *
+ * @param context main client context
+ */
+export default function NodeSyncManagerFactory(context) {
+
+  const producer = FullProducerFactory(context);
+  context.put(context.constants.PRODUCER, producer);
+
+  function startPolling() {
+    log.info('PUSH down or disconnected. Starting periodic fetch of data.');
+    if (!producer.isRunning())
+      producer.start(true); // `fetchers` are scheduled but not called immediately
+  }
+
+  function stopPollingAndSyncAll() {
+    log.info('PUSH (re)connected. Syncing and stopping periodic fetch of data.');
+    // if polling, stop
+    if (producer.isRunning())
+      producer.stop();
+    syncAll();
+  }
+
+  function syncAll() {
+    // fetch splits and segments. There is no need to catch this promise (it is handled by `SplitChangesUpdater`)
+    producer.synchronizeSplits().then(() => {
+      producer.synchronizeSegment();
+    });
+  }
+
+  let pushManager;
+
+  const settings = context.get(context.constants.SETTINGS);
+  if (settings.streamingEnabled)
```

Comment:
Consider marking the block ({}) if you're adding the line jumps next to the ifs everywhere. Or if you want use them as one liners.